pkg/activator/net/helpers.go

```diff
@@ -28,27 +28,31 @@ import (
 
 // endpointsToDests takes an endpoints object and a port name and returns a list
 // of l4 dests in the endpoints object which have that port
-func endpointsToDests(endpoints *corev1.Endpoints, portName string) sets.String {
-	ret := sets.NewString()
+func endpointsToDests(endpoints *corev1.Endpoints, portName string) destSets {
+	readyDests := sets.NewString()
+	notReadyDests := sets.NewString()
 
 	for _, es := range endpoints.Subsets {
 		for _, port := range es.Ports {
 			if port.Name == portName {
 				portStr := strconv.Itoa(int(port.Port))
 				for _, addr := range es.Addresses {
 					// Prefer IP as we can avoid a DNS lookup this way.
-					ret.Insert(net.JoinHostPort(addr.IP, portStr))
+					readyDests.Insert(net.JoinHostPort(addr.IP, portStr))
 				}
 				// Append not ready addresses because we want to send requests to
 				// pods before kube knows they are ready
 				for _, addr := range es.NotReadyAddresses {
-					ret.Insert(net.JoinHostPort(addr.IP, portStr))
+					notReadyDests.Insert(net.JoinHostPort(addr.IP, portStr))
 				}
 			}
 		}
 	}
 
-	return ret
+	return destSets{
```

Comment:
go just permits you to return two objects, wouldn't it be better/

Reply:
I think a struct is more convenient when it comes to sending these two lists through a channel, otherwise we will have to create something like this for the channel to receive these two lists `rwCh.ch <- (func() (sets.String, sets.String) { return readyDests, notReadyDests })` right?

Reply:
Hmmm. I guess for passing on the channel it might be useful :-)